src/sentry/sentry_metrics/indexer/cache.py

```diff
@@ -150,14 +160,28 @@ def get_many(self, namespace: str, keys: Iterable[str]) -> MutableMapping[str, O
     def set_many(self, namespace: str, key_values: Mapping[str, int]) -> None:
         cache_key_values = {self._make_cache_key(k): v for k, v in key_values.items()}
         self.cache.set_many(cache_key_values, timeout=self.randomized_ttl, version=self.version)
+        if options.get("sentry-metrics.indexer.write-new-cache-namespace"):
+            timestamp = int(datetime.utcnow().timestamp())
+            name_sapced_cache_key_values = {
+                self._make_namespaced_cache_key(namespace, k): self._make_cache_val(v, timestamp)
+                for k, v in key_values.items()
+            }
+            self.cache.set_many(
+                name_sapced_cache_key_values, timeout=self.randomized_ttl, version=self.version
+            )
 
     def delete(self, namespace: str, key: str) -> None:
-        cache_key = self._make_cache_key(key)
-        self.cache.delete(cache_key, version=self.version)
+        self.cache.delete(self._make_cache_key(key), version=self.version)
+        if options.get("sentry-metrics.indexer.write-new-cache-namespace"):
+            self.cache.delete(self._make_namespaced_cache_key(namespace, key), version=self.version)
 
     def delete_many(self, namespace: str, keys: Sequence[str]) -> None:
-        cache_keys = [self._make_cache_key(key) for key in keys]
-        self.cache.delete_many(cache_keys, version=self.version)
+        self.cache.delete_many([self._make_cache_key(key) for key in keys], version=self.version)
+        if options.get("sentry-metrics.indexer.write-new-cache-namespace"):
```

Comment:
This string option name is being used at a lot of places in the code. Same for read new cache namespace. Lets define string constants for these to avoid any possibility of copy/paste mishaps.

Reply:
Good point, fixed, thank you!